### DISCUSSION TITLE
improve the sql expression - and use LINQ as a base

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [vlang/v#11798] [Ideas]
I was talking to a friend of mine who mentioned he'd gotten into .NET lately. In that discussion, we talked about just how poerful LINQ is. I know that V already has `sql sqlConnection { dsl }` - but is that specific to just sqlite? I saw it show up in a few examples here and there and this, together with remembering how great LINQ was, made me wonder: What if V had something similiar?

First, reference: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/

Second, an example:
```v
fn main() {
  scores := [92, 96, 84]
  query := query {
    from s in scores
    where s < 90
    yield s
  }
  for row in query {
    println("This score is <90 : $row")
  }
}
```

This could be translated into something like this on the cgen side:

```c
// Approximation. Not what cgen actually generates.
query__iterableable query__new(query__QueryCommands[]);

void main__main__scores_q(query__QueryCtx* ctx) {
  // Rebuild the DSL into function calls and set the ctx.
}
void main__main() {
  int[] scores = [92, 96, 84];
  query = query__new(main__main__scores_q);
  while(query__has_next(query)) {
    int row = query__get_yield(query);
    printf("This score is <90 : %s\n", row);
  }
}
```

Basically, the query commands would be translated to C functions that act as callbacks to actually execute the query on a context, whilst in V it looks quite reminicent of a typical SQL syntax, but enhanced with some of the keywords only found in LINQ.

Since the SQL expression/DSL already exists, it'd be neat to make this more universal to traverse bigger objects like multi-dimensional maps or huge lists with a couple of selectors to them.

Now - this is just an idea I am throwing out there that I came up with at the spot. Nothing serious, nothing major. But I thought I'd still put it out anyway. :)

**Comment by IngwiePhoenix** [comment]
Additional examples. This also covers the method-style of LINQ of chaining methods together. However, this is - currently - not a solution for V as the lambda expression isn't quite there yet. https://www.tutorialsteacher.com/linq/sample-linq-queries
